### ISSUE TITLE
Add boolean arg `interger` to `mo.ui.number`

### ISSUE THREAD

**Issue by metaboulie** [marimo-team/marimo#2789]
### Description

In certain use cases with `mo.ui.number`, specifically when setting a random seed or key, some libraries require that this value be in integer form. To facilitate compatibility, adding a boolean argument, `integer`, could enable users to enforce integer formatting directly within `mo.ui.number`. 

Other potential applications for this enhancement include specifying the dimensional attributes of tensors, where data type consistency is often essential.

### Proposed Improvements
1. **Boolean `integer` Argument**: Add a boolean `integer` argument to enforce integer formatting for compatibility with libraries that require integers for specific functionalities like seeding.

2. **`dtype` Argument**: Introduce a `dtype` argument that allows explicit specification of the data type, enabling greater flexibility and alignment with various libraries that may have differing requirements (e.g., PyTorch, NumPy, TensorFlow).

3. **`format` Function for Custom Transformations**: Add a `format` function parameter that accepts a callable, which can automatically convert `number` to a target data type or structure. For example, using `lambda x: torch.tensor(x)` would set the value of `number` to a `torch.tensor`. This could be extended to other transformation functions based on the needs of the user.

### Alternatives
- **`dtype` Argument**: Incorporate a `dtype` argument that allows users to control the data type explicitly, providing enhanced compatibility with libraries that rely on specific data types (e.g., `torch.int`, `np.float32`).
- **Dynamic `format` Function for Transformations**: Implement a `format` parameter that accepts a callable function for dynamically converting `number` into any target type, like `lambda x: torch.tensor(x)`, allowing for seamless integration with libraries like PyTorch, TensorFlow, or JAX. This approach would give users more control over type conversions without needing to manage type checks or conversions manually.

**Comment by mscolnick** [comment]
This is already possible with, step=1:

```python
import marimo as mo
mo.ui.number(step=1)
```

We could update the docs to make this clearer
